bindings/go/oci/blob/artifact_blob.go

```diff
@@ -42,45 +52,48 @@ func NewResourceBlobWithMediaType(resource *descriptor.Resource, b blob.ReadOnly
 		}
 	}
 
-	if resource.Digest == nil {
-		if digAware, ok := b.(blob.DigestAware); ok {
-			if dig, ok := digAware.Digest(); ok {
-				digSpec, err := digestSpec(dig)
-				if err != nil {
-					return nil, fmt.Errorf("failed to parse digest spec from blob: %w", err)
+	if resource, ok := artifact.(*descriptor.Resource); ok {
```

Comment:
This is a bit weird... do I see it correctly that this type assert is in here twice? 🤔

Reply:
It was like that yes, but I changed the logic now so this no longer occurs.